opentelemetry-sdk/src/opentelemetry/sdk/_metrics/_internal/_view_instrument_match.py

```diff
@@ -124,46 +114,15 @@ def consume_measurement(self, measurement: Measurement) -> None:
 
         self._attributes_aggregation[attributes].aggregate(measurement)
 
-    def collect(self) -> Iterable[Metric]:
+    def collect(
+        self, aggregation_temporality: AggregationTemporality
+    ) -> Iterable[DataPointT]:
 
+        data_points = []
         with self._lock:
-            for (
-                attributes,
-                aggregation,
-            ) in self._attributes_aggregation.items():
-
-                previous_point = self._attributes_previous_point.get(
-                    attributes
-                )
-
-                current_point = aggregation.collect()
-
-                # pylint: disable=assignment-from-none
-
-                self._attributes_previous_point[
-                    attributes
-                ] = _convert_aggregation_temporality(
-                    previous_point,
-                    current_point,
-                    AggregationTemporality.CUMULATIVE,
+            for aggregation in self._attributes_aggregation.values():
+                data_points.append(
+                    aggregation.collect(aggregation_temporality)
```

Comment:
I read the previous conversation with Srikanth, but imo yield would still be nicer here

> when such list may not be iterated completely

You still avoid creating intermediate lists if you use a generator, so you don't have to allocate that memory just to free it. It also makes the code cleaner imo

Reply:
All right, changed to a generator :v: